src/components/Leaderboard.vue

```diff
@@ -0,0 +1,176 @@
+<script setup lang="ts">
+import { useAsyncState, useConfirmDialog } from '@vueuse/core'
+import { Leaderboards } from '../lib/api/Leaderboards'
+import { useSessionToken } from '../composables/useSessionToken'
+import { useAuth } from '../composables/useAuth'
+
+const props = defineProps<{
+	id: number
+}>()
+
+const token = useSessionToken()
+
+const leaderboards = new Leaderboards({
+	baseUrl: import.meta.env.VITE_BACKEND_URL
+})
+
+const {
+	state: board,
+	error,
+	isLoading,
+	execute
+} = useAsyncState(async () => {
+	// TODO: Add param in BE that allows also fetching deleted boards
+	const resp = await leaderboards.getLeaderboard(props.id)
+	return resp.data
+}, null)
+
+const {
+	reveal: revealDelete,
+	isRevealed: isRevealedDelete,
+	cancel: cancelDelete
+} = useConfirmDialog()
+
+const {
+	reveal: revealRestore,
+	isRevealed: isRevealedRestore,
+	cancel: cancelRestore
+} = useConfirmDialog()
+
+async function confirmDeleteBoard() {
+	// TODO: Error-handling
+	await leaderboards.deleteLeaderboard(props.id, useAuth(token.value))
+	execute()
+}
+
+async function confirmRestoreBoard() {
+	// TODO: Error-handling
+	await leaderboards.restoreLeaderboard(props.id, useAuth(token.value))
+	execute()
+}
+</script>
+
+<template>
+	<div class="container">
+		<div v-if="isLoading">Loading...</div>
+		<div v-else-if="error" class="error-container">
+			<p>Error</p>
+			<button @click="execute()" class="button">Reload</button>
+			<!-- TODO: Add button to go back to the List page -->
+		</div>
+
+		<div v-else class="main-content">
+			<!-- Deletion confirmation -->
+			<div v-if="isRevealedDelete" class="delete-confirmation-container">
+				<button @click="cancelDelete">Cancel</button>
+				<button @click="confirmDeleteBoard">Confirm</button>
+			</div>
+
+			<!-- Restoration confirmation -->
+			<div v-if="isRevealedRestore" class="delete-confirmation-container">
+				<button @click="cancelRestore">Cancel</button>
+				<button @click="confirmRestoreBoard">Confirm</button>
+			</div>
```

Comment:
We wanna make the confirmations modals after all, or you okay with them occupying space right below edit/delete buttons?

Reply:
I don't have too strong of an opinion. Even just a window alert is fine.

Reply:
Making em modals for two reasons:
1. I wanna learn how to make em
2. Already using `useConfirmDialog` and I don't wanna delete it all lmao